src/evict/evict_lru.c

```diff
@@ -1987,6 +1995,233 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
+        return;
+    }
+
+    /*
+     * Don't attempt eviction of internal pages with children in cache (indicated by seeing an
+     * internal page that is the parent of the last page we saw).
+     *
+     * Also skip internal page unless we get aggressive, the tree is idle (indicated by the tree
+     * being skipped for walks), or we are in eviction debug mode. The goal here is that if trees
+     * become completely idle, we eventually push them out of cache completely.
+     */
+    if (!FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_EVICT_AGGRESSIVE_MODE) &&
+      F_ISSET(ref, WT_REF_FLAG_INTERNAL)) {
+        if (page == last_parent) {
+            WT_STAT_CONN_INCR(session, cache_eviction_server_skip_intl_page_with_active_child);
+            return;
+        }
+        if (__wt_atomic_load32(&btree->evict_walk_period) == 0 && !__wt_cache_aggressive(session))
+            return;
+    }
+
+    /* If eviction gets aggressive, anything else is fair game. */
+    if (__wt_cache_aggressive(session))
+        goto fast;
+
+    if (modified && __evict_skip_dirty_candidate(session, page))
+        return;
+
+fast:
+    /* If the page can't be evicted, give up. */
+    if (!__wt_page_can_evict(session, ref, NULL))
+        return;
+
+    WT_ASSERT(session, evict->ref == NULL);
+    if (!__evict_push_candidate(session, queue, evict, ref))
+        return;
+
+    *queuedp = true;
+    __wt_verbose(session, WT_VERB_EVICTSERVER, "walk select: %p, size %" WT_SIZET_FMT, (void *)page,
+      __wt_atomic_loadsize(&page->memory_footprint));
+
+    return;
+}
+
+/*
+ * __evict_walk_tree --
+ *     Get a few page eviction candidates from a single underlying file.
+ */
+static int
+__evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_entries, u_int *slotp)
+{
+    WT_BTREE *btree;
+    WT_CACHE *cache;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_EVICT_ENTRY *end, *evict, *start;
+    WT_PAGE *last_parent, *page;
+    WT_REF *ref;
+    WT_TXN *txn;
+    uint64_t internal_pages_already_queued, internal_pages_queued, internal_pages_seen;
+    uint64_t min_pages, pages_already_queued, pages_queued, pages_seen, refs_walked;
+    uint32_t evict_walk_period, target_pages, walk_flags;
+    int restarts;
+    bool give_up, queued, urgent_queued;
+
+    conn = S2C(session);
+    btree = S2BT(session);
+    cache = conn->cache;
+    last_parent = NULL;
+    restarts = 0;
+    give_up = urgent_queued = false;
+    txn = session->txn;
+
+    WT_ASSERT_SPINLOCK_OWNED(session, &cache->evict_walk_lock);
+
+    start = queue->evict_queue + *slotp;
+    target_pages = __evict_get_target_pages(session, max_entries, slotp);
+
+    /* If we don't want any pages from this tree, move on. */
+    if (target_pages == 0)
+        return (0);
+
+    end = start + target_pages;
+
+    min_pages = __evict_get_min_pages(session, target_pages);
+
+    WT_RET(__evict_walk_prepare(session, &walk_flags));
```

Comment:
`__evict_walk_prepare` is allowed to return `WT_NOTFOUND` as a legal value. `WT_RET` will error out in a previously legal case:
```suggestion
    ret = __evict_walk_prepare(session, &walk_flags);
    WT_RET_NOTFOUND_OK(ret);
```

Reply:
good point. Thanks!